MCDataProducts/inc/SimParticle.hh

```diff
@@ -229,7 +231,7 @@ namespace mu2e {
     std::vector<key_type>                      daughterIds() const;
 
     // Is the second half defined?
-    bool endDefined() const { return _endDefined;}
+    bool endDefined() const { return _stoppingCode != ProcessCode::unknown; }
```

Comment:
Potentially risky. I do not fully remember all the details, but believe that it can happen that it may end up that way in a "defined end", consider leaving is as it was, also for a potential future use.

Reply:
That bool member was set to 'true' in the same function where the stoppingcode was set, thus the only way a SimParticle could have _endDefined ==true and stoppingCode == unknown is if the process which stopped the particle was itself 'unknown' (ie default value of ProcessCode) .  It seems wrong that any actual process should use code 'uknown', but If necessary, I will introduce a new code 'uninitialized' to distinguish these cases.

Reply:
That should be sufficient

Reply:
are you saying you think 'uninitialized' is needed?

Reply:
Yes, I was trying to agree with the introduction of a name/state different from/in addition to "unknown".